src/main/java/org/radarbase/ksql/udf/RestInferenceUdf.java

```diff
@@ -0,0 +1,134 @@
+package org.radarbase.ksql.udf;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.util.Map;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.apache.kafka.common.Configurable;
+import org.apache.log4j.BasicConfigurator;
+import org.radarbase.ksql.util.HttpClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+@UdfDescription(name = "api_inference",
+        author = "yatharthranjan",
+        version = "1.0.0",
+        description = "A custom function to run real time inference using REST API calls.")
+public class RestInferenceUdf implements Configurable {
+
+    private static final Logger logger = LoggerFactory.getLogger(RestInferenceUdf.class);
+
+    private final transient OkHttpClient httpClient;
+    private final transient ObjectMapper objectMapper;
+    private transient String apiUrl;
+
+    public RestInferenceUdf() {
+        httpClient = HttpClientFactory.getClient();
+        objectMapper = new ObjectMapper();
+        BasicConfigurator.configure();
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        apiUrl = (String) configs.get("ksql.functions.api_inference.base.url");
+    }
+
+    @Udf(
+            description = "Run inference in realtime by loading data from the database (i.e. we " +
+                    "only pass metadata as params). This will return the json response as-is in " +
+                    "String format. If needed, the values can be extracted in KSQL using " +
+                    "the 'EXTRACTJSONFIELD' scalar function."
+    )
+    public String runMetadataInference(
+            String dataLoaderModule,
+            String dataLoaderClass,
+            String dbName,
+            String projectId,
+            String userId,
+            String modelName,
+            String modelVersion,
+            String sourceId,
+            Double startTime,
+            Double endTime
+    ) {
+        if (dataLoaderModule==null
+                || dataLoaderClass==null
+                || dbName==null
+                || projectId==null
+                || userId==null
+                || modelName==null
+        ) {
+            logger.warn("One of the required parameters was null");
+            return null;
+        }
+
+        if (modelVersion==null) {
+            modelVersion = "best";
```

Comment:
When calling the best model, you can also add `metric` as a query to get the model whose performance is  best for that particular metric. It was useful in the case of having multiple metrics in a model.  If it is not given, it will use the first metric available.